prebid-server/pbs-modules/51degrees-device-detection.md

```diff
@@ -0,0 +1,351 @@
++                    "os": "Android",
++                    "osv": "11.0",
++                    "h": 3200,
++                    "w": 1440,
++                    "ext": {
++                        "fiftyonedegrees_deviceId": "110698-102757-105219-0"
++                    }
+                 },
+```
+
+[Enterprise](https://51degrees.com/pricing) files can provide even more information:
+
+```diff
+                 "device": {
+                     "ua": "Mozilla/5.0 (Linux; Android 11; SM-G998W) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/86.0.4240.198 Mobile Safari/537.36",
++                    "devicetype": 1,
++                    "make": "Samsung",
++                    "model": "SM-G998W",
+                     "os": "Android",
+                     "osv": "11.0",
+                     "h": 3200,
+                     "w": 1440,
++                    "ppi": 516,
++                    "pxratio": 3.44,
+                     "ext": {
+-                        "fiftyonedegrees_deviceId": "110698-102757-105219-0"
++                        "fiftyonedegrees_deviceId": "110698-102757-105219-18092"
+                     }
+```
+
+## Running the demo (PBS-Go)
+
+1. Download dependencies:
+
+```bash
+go mod download
+```
+
+{:start="2"}
+2. Replace the original config file `pbs.json` (placed in the repository root or in `/etc/config`) with the sample [config file](https://github.com/51Degrees/prebid-server/blob/feature/51Degrees-device-detection-module/modules/fiftyone_degrees/device_detection/sample/pbs.json):
+
+```bash
+cp modules/fiftyone_degrees/device_detection/sample/pbs.json pbs.json
+```
+
+{:start="3"}
+3. Download `51Degrees-LiteV4.1.hash` from [[GitHub](https://github.com/51Degrees/device-detection-data/blob/main/51Degrees-LiteV4.1.hash)] and put it in the project root directory.
+
+```bash
+curl -o 51Degrees-LiteV4.1.hash -L https://github.com/51Degrees/device-detection-data/raw/main/51Degrees-LiteV4.1.hash
+```
+
+{:start="4"}
+4. Create a directory for sample stored requests (needed for the server to run):
+
+```bash
+mkdir -p sample/stored
+```
+
+{:start="5"}
+5. Start the server:
+
+```bash
+go run main.go
+```
+
+{:start="6"}
+6. Run sample request:
+
+```bash
+curl \
+--header "Content-Type: application/json" \
+http://localhost:8000/openrtb2/auction \
+--data @modules/fiftyone_degrees/device_detection/sample/request_data.json
+```
+
+## Maintainer contacts
+
+Any suggestions or questions can be directed to [support@51degrees.com](mailto:support@51degrees.com) e-mail.
+
+Or just open new [issue](https://github.com/prebid/prebid-server-java/issues/new) or [pull request](https://github.com/prebid/prebid-server-java/pulls) in this repository.
+
+## Further Reading
+
+* [Prebid Server Module List](/prebid-server/pbs-modules/index.html)
+* [Building a Prebid Server Module](/prebid-server/developers/add-a-module.html)
```

Comment:
Would suggest removing this last row - not clear how it's useful for users of your module to have a link to build their own module.

Reply:
removed, thanks